frontend/src/components/UpgradeDialog.tsx

```diff
@@ -0,0 +1,122 @@
+import { LifeBuoy, Mail, RefreshCw, Users, Zap } from "lucide-react";
+import { ReactNode } from "react";
+import { Button } from "src/components/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "src/components/ui/dialog";
+import { ALBY_PRO_PLAN } from "src/constants";
+import { useAlbyMe } from "src/hooks/useAlbyMe";
+import { openLink } from "src/utils/openLink";
+
+interface UpgradeDialogProps {
+  children: ReactNode;
+}
+
+export const UpgradeDialog = ({ children }: UpgradeDialogProps) => {
```

Comment:
As a follow up, I think we could consider Alby Cloud too. Right now it's hardcoded to the pro plan.

We have a menu item called "Alby Cloud" on the sidebar when the user is logged in but does not have a plan - should we change anything here? why does it only show when logged in for example? (I think this no longer applies?)

Reply:
Good point, we decided to have a general "upgrade" button in there for now and allow users to decide between diy + cloud themselves for now. (since for desktop users a hosted option might also be relevant)